### ISSUE TITLE
Allow editing overwrites through channel.edit()

### ISSUE THREAD

**Issue by jack1142** [Rapptz/discord.py#2197]
### The Problem

This will allow to update all overwrites of a channel at the same time.

### The Ideal Solution

`overwrites` kwarg in `channel.edit()` which will work the same as the one used in `Guild.create_text_channel()`
Discord API supports updating all overwrites of a channel at the same time using the very same endpoint: https://discordapp.com/developers/docs/resources/channel#modify-channel

### The Current Solution

Using `channel.set_permissions()` for each overwrites target separately, which requires more http requests.

### Summary

Basically this:
```py
overwrites = {
    guild.default_role: discord.PermissionOverwrite(read_messages=False),
    guild.me: discord.PermissionOverwrite(read_messages=True)
}

await channel.edit(overwrites=overwrites)
```

**Comment by jack1142** [comment]
Looks like #2198 resolved this, closing.
